File: go/pkg/stdin/stdin.go
package stdin

import (
	"bufio"
	"log"
	"os"
)

func GetScanner() *bufio.Scanner {
	info, err := os.Stdin.Stat()
	if err != nil {
		log.Fatal(err)
	}

	if info.Mode()&os.ModeCharDevice != 0 {
		log.Fatal("Input data must be piped in via stdin")
	}

	return bufio.NewScanner(os.Stdin)
}

File: deno/lib/stdin.ts
import { TextLineStream } from "@std/streams";

if (Deno.stdin.isTerminal()) {
  throw Error("Input data must be piped in via stdin");
}

export const lineStream = Deno.stdin.readable
  .pipeThrough(new TextDecoderStream())
  .pipeThrough(new TextLineStream());
